pypesto/optimize/util.py

```diff
@@ -0,0 +1,68 @@
+from ..engine import Engine, SingleCoreEngine
+from ..objective import HistoryOptions
+from ..store.save_to_hdf5 import get_or_create_group
+from pathlib import Path
+
+
+import h5py
+
+
+def check_hdf5_mp(history_options: HistoryOptions,
+                  engine: Engine):
+    """
+    Create a folder for partial HDF5 files,
+    if a parallelization engine will be used.
+
+    Paramters
+    ---------
+    engine:
+        The Engine which is used in the optimization
+    history_options:
+        The HistoryOptions used in the optimization
+
+    Returns
+    -------
+    The filename that will be used to combine the partial HDF5 files later.
+    If a parallelization engine is not used, `None` is returned.
+    """
+    if not isinstance(engine, SingleCoreEngine):
+        filename = history_options.storage_file
+        file_path = Path(filename)
+        # create directory with same name as original file stem
+        partial_file_path = (
+            file_path.parent / file_path.stem /
+            (file_path.stem + '_{id}' + file_path.suffix)
+        )
+        partial_file_path.parent.mkdir(parents=True, exist_ok=True)
+        history_options.storage_file = str(partial_file_path)
+        # create hdf5 file that gathers the others within history group
+        with h5py.File(filename, mode='a') as f:
+            get_or_create_group(f, "history")
+        return filename
+    return None
+
+
+def fill_hdf5_file(ret,
+                   filename):
```

Comment:
type hints for inputs missing

Reply:
anything I could put into ret? I think it depends on the engine what is returned here.